api/wasm/cpp/proxy_wasm_enums.h

```diff
@@ -0,0 +1,13 @@
+/*
```

Comment:
Should this interface tree be versioned? I.e. the C++ ABI might have a v2 in the future, it should have a directory structure and tree that reflects this.

Reply:
We consider this to be v0 since we we haven't gotten it all in and gotten buy-in from the community.  I'll move this to api/wasm/cpp/v0/.... and we can snap v1 when we have upstreamed and addressed any pending issues (e.g. renaming).

Reply:
Do you plan to use semantic version, or a scheme similar to our APIs, e.g. v1alpha, v2, v2alpha, etc?

Reply:
Semantic versioning is probably overkill at this point, particularly since WASI hasn't (AFAICT) decided on how they will do versioning in the long run.

Reply:
I really think we should mostly ignore WASI right now. The WASM ecosystem is too immature to be focused on standardization in this first pass. Can we just go ahead and do a v0 for now? I agree that full semantic versioning might be a bit premature.